### PR TITLE
Implements modal according to design specs

### PR DESCRIPTION
**Pull request by Kushagra8888**
Refer: https://github.com/adwyze/KnitUI/pull/15

- Also adds animation on enter based on https://www.notion.so/Design-handoff-f946cd7080be4215b7195495acf7a562

### CODE REVIEW

**Comment by Rohanhacker** [comment]
export it from the main components/index.tsx

File: components/Modal/ModalWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+import { Icon } from "../Icon"
+
+const sizeToWidth = {
+  "small": "49rem",
+  "medium": "63rem",
+  "large": "91rem",
+  "x-large": "105rem"
+}
+
+export interface ModalWrapperProps {
```
[the remark this answers is not in the document]
Comment (Kushagra8888) [reply components/Modal/ModalWrapper.tsx]: Will be adding 1, but I think we should restrict 2 and 3 to enforce consistency of design

**Comment by arorayash** [comment]
Screenshot 2019-05-31 at 12 28 45 PM

- The close button in some stories is there on the right side of the header, when it should be fixed on the top-right. 
- The header in the overflowing content ones doesn't look quite right. the bottom padding is missing?
- Can we have a story for different sizes?
- Also, I'm considering to have another variation for overall padding. Will get a better idea checking story on different sizes.

@Kushagra8888 please let me know if there's anything missing.

**Comment by Kushagra8888** [comment]
@arorayash You can change the sizes using the "Knobs "panel.

For the close button, in our last discussion we concluded that the close icon should be fixed to the top right corner only when the header contains a right section, and placed in the right otherwise. (We had decided on heder containing a title section and a right section). Let me know if this is not how it is supposed to behave...

I'll check the header in the overflow case.

**Comment by arorayash** [comment]
Okay yeah, tried the different sizes, works great! @Kushagra8888. Will need to add another variation of padding. Let me get back to you with the design.

In the last discussion, we decided that the close button will always be fixed on top. 🙊 And for the same reason, the other case(of having the close on the right) is not there on Figma anymore.

**Comment by Kushagra8888** [comment]
@arorayash Ohh, I must have forgot about that... will change it...

**Comment by anshumanv** [comment]
Conflicts

**Comment by Kushagra8888** [comment]
@arorayash I have added animations on openning, but on leave, they do not seem to be working, I have not dug too deep into it, but it seems to be a problem introduced with Portals in React 16.
https://github.com/adwyze/KnitUI/pull/33/files#diff-5d2ad276e9ca9e82acaea710e300d53bR101

**Pull request merged**